File: NextSpring/src/main/java/com/study/nextspring/domain/post/entity/PostComment.java
package com.study.nextspring.domain.post.entity;

import com.study.nextspring.domain.member.entity.Member;
import com.study.nextspring.global.exception.ServiceException;
import com.study.nextspring.global.jpa.entity.BaseTime;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostComment extends BaseTime {
    @ManyToOne(fetch = FetchType.LAZY)
    private Post post;

    @ManyToOne(fetch = FetchType.LAZY)
    private Member author;

    @Column(columnDefinition = "TEXT")
    private String content;

    public void modify(String content) {
        this.content = content;
    }

    public void checkActorCanModify(Member actor) {
        if (actor == null) throw new ServiceException("401-1", "로그인 후 이용해주세요.");

        if (actor.equals(author)) return;

        throw new ServiceException("403-2", "작성자만 댓글을 수정할 수 있습니다.");
    }

    public void checkActorCanDelete(Member actor) {
        if (actor == null) throw new ServiceException("401-1", "로그인 후 이용해주세요.");

        if (actor.isAdmin()) return;

        if (actor.equals(author)) return;

        throw new ServiceException("403-2", "작성자만 댓글을 삭제할 수 있습니다.");
    }
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/domain/comment/entity/PostComment.kt
package com.krstudy.kapi.com.krstudy.kapi.domain.comment.entity

import com.krstudy.kapi.com.krstudy.kapi.domain.member.entity.Member
import com.krstudy.kapi.com.krstudy.kapi.domain.post.entity.Post
import com.krstudy.kapi.com.krstudy.kapi.global.jpa.BaseEntity
import jakarta.persistence.*

@Entity
class PostComment private constructor(
    @ManyToOne
    var author: Member? = null,

    @ManyToOne
    var post: Post? = null,

    @Column(columnDefinition = "TEXT")
    var body: String? = null
) : BaseEntity() {
    companion object {
        fun builder() = Builder()
    }

    class Builder {
        private var author: Member? = null
        private var post: Post? = null
        private var body: String? = null

        fun author(author: Member) = apply { this.author = author }
        fun post(post: Post) = apply { this.post = post }
        fun body(body: String) = apply { this.body = body }
        fun build() = PostComment(author, post, body)
    }
}